x-pack/qa/ml-native-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ForecastIT.java

```diff
@@ -216,36 +216,59 @@ public void testMemoryLimit() throws Exception {
         analysisConfig.setBucketSpan(bucketSpan);
         DataDescription.Builder dataDescription = new DataDescription.Builder();
         dataDescription.setTimeFormat("epoch");
-        Job.Builder job = new Job.Builder("forecast-it-test-memory-limit");
+        Job.Builder job = new Job.Builder("forecast-it-test-overflow-to-disk");
+        AnalysisLimits limits = new AnalysisLimits(2048L, null);
+        job.setAnalysisLimits(limits);
         job.setAnalysisConfig(analysisConfig);
         job.setDataDescription(dataDescription);
 
         registerJob(job);
         putJob(job);
         openJob(job.getId());
         createDataWithLotsOfClientIps(bucketSpan, job);
-        ElasticsearchException e = expectThrows(ElasticsearchException.class,
-                () -> forecast(job.getId(), TimeValue.timeValueMinutes(120), null));
-        assertThat(e.getMessage(),
-                equalTo("Cannot run forecast: Forecast cannot be executed as forecast memory usage is predicted to exceed 20MB"));
+
+        try {
+            String forecastId = forecast(job.getId(), TimeValue.timeValueHours(1), null);
+
+            waitForecastToFinish(job.getId(), forecastId);
+        } catch (ElasticsearchStatusException e) {
+            if (e.getMessage().contains("disk space")) {
+                throw new ElasticsearchStatusException(
+                        "Test likely fails due to insufficient disk space on test machine, please free up space.", e.status(), e);
```

Comment:
Do you know how much free space a test machine has to have to avoid this situation?  If it's 1GB that's probably OK, but if it's 5GB then this could annoy people if a multi-hour test run fails because of it.

Reply:
Yes, if the diskspace is below 5GB the folder for tmp storage will not be injected, which will cause the engine to fail.

I only see 2 options:

 - I lower the limit, 5GB is still quite large, but I would not go below 2GB.
 - I drop the test and we cover this as QA test

I do not think 5GB is that much of a problem these days, only if run in a virtual environment where virtual disks are small in size.

Do we have any information about diskspace usage of other integration tests? Indexes can get quite large as well.

Reply:
> only if run in a virtual environment where virtual disks are small in size

This is the situation I was thinking of.  Tests are quite often run in VMs to test different operating systems etc.

I think it would be good to cover this as a QA test too, but maybe a third option would be instead of throwing an exception if there's a disk space error, use the "assume" feature if the free disk space in the build directory is less than 5GB?  Something along the lines of this (but with the correct code for finding available disk space substituted):

```
    if (e.getMessage().contains("disk space")) {
        assumeTrue("Insufficient disk space to run test", something.getDiskSpaceInGB() > 5);
    }
```

If a test fails due to an assumption not holding then it does _not_ fail the entire test suite.

Reply:
I thought about this as well, but have bad feelings as this is like disabling the test, because no one reads the log, no one will notice if this starts failing.

To throw in another option: we could disable the 5 GB check, for the test as I need much less (the models use 77MB heap size, the json is larger, but for sure below 500MB). To disable it, there could be a `force` parameter in the forecast action or we make the 5GB a `node parameter` and tweak it in the test setup.

Reply:
@droberts195 Can you take a quick look at https://github.com/elastic/elasticsearch/pull/30399/commits/315fbde9f632545f5cea86815d030c289aa03d50?